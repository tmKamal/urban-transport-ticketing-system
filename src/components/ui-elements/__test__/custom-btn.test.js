import React from 'react';
import ReactDom from 'react-dom';
import CustomBtn from '../custom-btn';
import { getByTestId, render } from '@testing-library/react';
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer';

it("renders without crashing",()=>{
    const div=document.createElement("div");
    ReactDom.render(<CustomBtn></CustomBtn>,div);
}); // checks that the element renders without crashing (-- Test case 04 --)

it("renders custom btn as it should be with name prop",()=>{
    const {getByTestId}=render(<CustomBtn name="test 02"></CustomBtn>);
    expect(getByTestId("cbtn")).toHaveTextContent("test 02");
}); // checks that given props functions as they should (-- Test case 05 --)

it("renders custom btn as it should be with name prop",()=>{
    const {getByTestId}=render(<CustomBtn name="test 02 dif"></CustomBtn>);
    expect(getByTestId("cbtn")).toHaveTextContent("test 02 same");
}); // checks that given props functions as they should (-- Test case 06 --) [Negative]



it("matches the snapshot",()=>{
    const tree=renderer.create(<CustomBtn name="test 02"></CustomBtn>).toJSON();
    expect(tree).toMatchSnapshot();
}); // Snap shot matcher
