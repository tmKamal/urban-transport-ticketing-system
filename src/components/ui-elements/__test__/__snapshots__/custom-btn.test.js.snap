// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-text MuiButton-textPrimary"
  disabled={false}
  name="test 02"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
>
  <span
    className="MuiButton-label"
  />
</button>
`;
